src/test/shared/cloudformation/templateRegistry.test.ts

```diff
@@ -0,0 +1,140 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as path from 'path'
+import * as vscode from 'vscode'
+
+import { CloudFormationTemplateRegistry, TemplateData } from '../../../shared/cloudformation/templateRegistry'
+import { rmrf } from '../../../shared/filesystem'
+import { makeTemporaryToolkitFolder } from '../../../shared/filesystemUtilities'
+import { assertThrowsError } from '../utilities/assertUtils'
+import { badYaml, makeSampleSamTemplateYaml, strToYamlFile } from './cloudformationTestUtils'
+
+describe('CloudFormation Template Registry', async () => {
+    const goodYaml1 = makeSampleSamTemplateYaml(false)
+    const goodYaml2 = makeSampleSamTemplateYaml(true)
+
+    describe('CloudFormationTemplateRegistry', async () => {
+        const testRegistry: CloudFormationTemplateRegistry = CloudFormationTemplateRegistry.getRegistry()
```

Comment:
These tests are intended to exercise portions of the CloudFormationTemplateRegistry class. The singleton instance is not of interest to the tests. Using shared data for multiple tests is a frequent cause of test instability.

Instantiate a new CloudFormationTemplateRegistry object for each test. Do not use the singleton instance.